lib/earmark_parser/line_scanner.ex

```diff
@@ -303,8 +307,10 @@ defmodule EarmarkParser.LineScanner do
   defp _split_table_columns(line) do
     line
     |> String.split(~r{(?<!\\)\|})
-    |> Enum.map(&String.trim/1)
-    |> Enum.map(fn col -> Regex.replace(~r{\\\|}, col, "|") end)
+    |> Enum.map(fn col ->
```

Comment:
This is an optimisation which is not in the spirit of the rest of the code of your PR.
I fail to succeed in imagining a table in markdown with more then 1000 lines and that is kind of the magnitude I start worrying about the constant in front of the _n_

Reply:
... which is very very cautios

Reply:
An alternative would be to use `Stream.map` but I guess this would be slower for _n_ in this kind of ranges, so I'll keep it simple.

Reply:
I agree that this was a bit of an outlier. I learn a lot with each PR in terms of all sorts of considerations. I've gone ahead and removed this.